portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -37,8 +37,18 @@
   
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Integer guaranteed in script.js
-    int limit = Integer.parseInt(request.getParameter("limit"));
+    int limit = 3;
+    try {
+      limit = Integer.parseInt(request.getParameter("limit"));
+    }
+    catch(NumberFormatException e){
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST,"Comment limit must be an integer.");
```

Comment:
You should say here that it should be a non-negative as well.

Reply:
Done.